tutorials/preprocessing/plot_35_artifact_correction_regression.py

```diff
@@ -0,0 +1,173 @@
+               eog=True, ecg=True, ref_meg=False).load_data()
+raw.info['bads'] = ['MLO52-4408', 'MRT51-4408', 'MLO42-4408', 'MLO43-4408']
+raw.filter(None, 40)
+decim = 12  # 2400 -> 200 Hz sample rate for epochs
+
+###############################################################################
+# Removing artifacts by regression
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# Generally speaking, artifacts whose time waveforms on the sensors are
+# accurately reflected by some reference signal can be removed by regression.
+# Two examples of this are blink artifacts (captured by EOG channels), and
+# heartbeat artifacts (captured by ECG electrodes).
+#
+#
+# Example: EOG artifacts
+# ~~~~~~~~~~~~~~~~~~~~~~
+#
+# First let's epoch our data the normal way, using just event type 1 (only the
+# "standard" trials) for simplicity. Event timing is adjusted by comparing the
+# trigger times on detected sound onsets on channel UADC001-4408.
+
+events = mne.find_events(raw, stim_channel='UPPT001')
+sound_data = raw[raw.ch_names.index('UADC001-4408')][0][0]
+onsets = np.where(np.abs(sound_data) > 2. * np.std(sound_data))[0]
+min_diff = int(0.5 * raw.info['sfreq'])
+diffs = np.concatenate([[min_diff + 1], np.diff(onsets)])
+onsets = onsets[diffs > min_diff]
+assert len(onsets) == len(events)
+events[:, 0] = onsets
+epochs = mne.Epochs(raw, events, event_id=1, decim=decim, preload=True)
+
+###############################################################################
+# Next we'll compare the `~mne.Evoked` data (average across epochs) before and
+# after we regress out the EOG signal. We do this by estimating the regression
+# coefficients on epoch data with the evoked response subtracted out, then use
+# those coefficients to remove the EOG signal from the original data:
+
+# do regression
+_, betas = mne.preprocessing.regress_artifact(epochs.copy().subtract_evoked())
+epochs_clean, _ = mne.preprocessing.regress_artifact(epochs, betas=betas)
+
+# get ready to plot
+plot_picks = ['meg', 'eog', 'ecg']
+evo_kwargs = dict(picks=plot_picks, spatial_colors=True,
+                  verbose='error')  # ignore warnings about spatial colors
+
+# plot original data (averaged across epochs)
+fig = epochs.average(picks=plot_picks).plot(**evo_kwargs)
+fig.suptitle('Auditory epochs')
+fig.tight_layout()
+
+# plot regressed data
+fig = epochs_clean.average(picks=plot_picks).plot(**evo_kwargs)
+fig.suptitle('Auditory epochs, EOG regressed')
+fig.tight_layout()
+
+# clean up
+del epochs, epochs_clean
+
+###############################################################################
+# The effect is subtle in these evoked data, but you can see that a bump toward
+# the end of the window has had its amplitude decreased.
+#
+# The effect is clearer if we create epochs around (autodetected) blink events,
+# and plot the average blink artifact before and after the regression (note
+# that the y-axis limits change between the left and right plots):
```

Comment:
I think it would be more impressive if the y-axis limit didn't change. Can we set it to the original limit so that it is very obvious that the eye blinks are gone (we're not really interested in the MEG signals so why zoom in).